docs/package_templates/meson_package/all/conanfile.py

```diff
@@ -0,0 +1,166 @@
+    homepage = "https://github.com/project/package"
+    # no "conan" and project name in topics. Use topics from the upstream listed on GH
+    topics = ("topic1", "topic2", "topic3")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                # once removed by config_options, need try..except for a second del
+                del self.options.fPIC
+            except Exception:
+                pass
+        # for plain C projects only
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        self.requires("dependency/0.8.1")
+
+    def validate(self):
+        # validate the minimum cpp standard supported. For C++ projects only
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support."
+                )
+        # in case it does not work in another configuration, it should validated here too
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    # if another tool than the compiler or Meson is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
```

Comment:
Should we add a `tool_requires` for `pkgconf`? A lot of Meson projects use `pkg-config` files for dependencies.

Reply:
This would result in an unneeded build requirement for projects that don't have dependencies, but it could be argued that pkgconf isn't burdensome to make available "for free".

Meson itself doesn't make that judgment call, it will only check for pkg-config if you try to look up dependencies.

Reply:
First, need to think, does it work on CCI? Those templates are mainly focused on CCI. Meson uses largely pkg-config no doubt.

Reply:
Since this is a generic template with a lot of extra I think it's totally fine to drop in. 👍 for the suggestion
Super specific templates would be hard to maintain

Reply:
I've added pkgconf, doing a quick look, most of recipes using Meson, also specify pkgconfig. Only Docker images have it installed, so Windows CI would fail. (Mac I didn't check).